runners/google-cloud-dataflow-java/src/test/java/org/apache/beam/runners/dataflow/DataflowPipelineJobTest.java

```diff
@@ -655,4 +659,71 @@ public void sleep(long millis) throws InterruptedException {
       fastNanoTime += millis * 1000000L + ThreadLocalRandom.current().nextInt(500000);
     }
   }
+
+  @Test
+  public void testCancelUnterminatedJobThatSucceeds() throws IOException {
+    Dataflow.Projects.Jobs.Update update = mock(Dataflow.Projects.Jobs.Update.class);
+    Job content = new Job();
+    content.setProjectId(PROJECT_ID);
+    content.setId(JOB_ID);
+    content.setRequestedState("JOB_STATE_CANCELLED");
+    when(mockJobs.update(eq(PROJECT_ID), eq(JOB_ID), eq(content))).thenReturn(update);
```

Comment:
When using "when" clauses they are not strict and use functions like any(Job.class)/anyString only using something specific if a different result needs to be returned based upon an input parameter. The "verify" clauses should be the strict ones.

This helps make tests not fail due to NPEs since a null would be returned if the mock call wasn't satisfied and will instead fail with a much cleaner message saying parameters X, Y, and Z were expected but got A, B, and C

This applies to this test and the ones you added below